framework/include/utils/PiecewiseByBlockLambdaFunctor.h

```diff
@@ -162,6 +166,11 @@ PiecewiseByBlockLambdaFunctor<T>::setFunctor(const MooseMesh & libmesh_dbg_var(m
 
   for (const auto block_id : block_ids)
     add_lammy(block_id);
+
+  // Handle special case of ANY_BLOCK_ID and empty block restriction that also cover
+  // INVALID_BLOCK_ID
+  if (block_ids.count(Moose::ANY_BLOCK_ID) || block_ids.empty())
+    add_lammy(Moose::INVALID_BLOCK_ID);
```

Comment:
For `ANY_BLOCK_ID` in particular, should we loop over the mesh subdomains and add the lambda for every block?